### PR TITLE
提交八个算子——修改补充

### PR DESCRIPTION
**Pull request by 1147022743**
### PR Docs

https://github.com/PaddlePaddle/docs/pull/6668
### PR APIs

torch.Tensor.select_scatter
torch.Tensor.slice_scatter
torch.tensor_split
torch.Tensor.tensor_split
torch.hsplit
torch.Tensor.hsplit
torch.dsplit
torch.Tensor.dsplit

### CODE REVIEW

**Comment by 1147022743** [comment]
> @1147022743 麻烦根据规范补充一下映射文档 https://github.com/PaddlePaddle/PaConvert/blob/master/docs/CONTRIBUTING.md#%E6%AD%A5%E9%AA%A42%E7%BC%96%E5%86%99api%E6%98%A0%E5%B0%84%E5%85%B3%E7%B3%BB ，每一个API都需要对应的映射文档
> 
> ```
> torch.Tensor.slice_scatter
> torch.tensor_split
> torch.Tensor.tensor_split
> torch.hsplit
> torch.Tensor.hsplit
> torch.dsplit
> torch.Tensor.dsplit
> ```


八个算子映射文档里都有

**Comment by zhwesky2010** [comment]
> > @1147022743 麻烦根据规范补充一下映射文档 https://github.com/PaddlePaddle/PaConvert/blob/master/docs/CONTRIBUTING.md#%E6%AD%A5%E9%AA%A42%E7%BC%96%E5%86%99api%E6%98%A0%E5%B0%84%E5%85%B3%E7%B3%BB ，每一个API都需要对应的映射文档
> > ```
> > torch.Tensor.slice_scatter
> > torch.tensor_split
> > torch.Tensor.tensor_split
> > torch.hsplit
> > torch.Tensor.hsplit
> > torch.dsplit
> > torch.Tensor.dsplit
> > ```
> 
> 八个算子映射文档里都有

在哪里，把这八个文档的链接贴出来

File: paconvert/api_matcher.py [hunk 1]
```diff
@@ -149,6 +149,64 @@ def generate_code(self, kwargs):
...
 
+class TensorSlice_scatterMatcher(BaseMatcher):
+    def generate_code(self, kwargs):
+        API_TEMPLATE = textwrap.dedent(
+            """
+            x = {}
+            shape = x.shape
+            axes, starts, strides = [0], [0], [1]
+            ends = shape[axes[0]]
+            """
```
Comment (zhwesky2010) [inline paconvert/api_matcher.py:160]:
```
shape = {}.shape
axes, starts, strides, ends = [0], [0], [1], shape[axes[0]]
```
四行变两行，写法越简洁越好

File: tests/test_Tensor_slice_scatter.py [hunk 2]
```diff
@@ -24,30 +24,56 @@ def test_case_1():
         """
         import torch
         a = torch.zeros(8, 8)
-        b = torch.ones(8)
-        result = a.slice_scatter(b, start=6)
+        b = torch.ones(2, 8)
+        result = a.slice_scatter(b, start = 6)
```
Comment (zhwesky2010) [inline tests/test_Tensor_slice_scatter.py:28]: 这个再加一点参数缺省的写法吧，可以没有 end、step、dim

File: tests/test_Tensor_tensor_split.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    )
+    obj.run(pytorch_code, ["result"])
+
+
+def test_case_4():
+    pytorch_code = textwrap.dedent(
+        """
+        import torch
+        a = torch.arange(14).reshape(2, 7)
+        result = a.tensor_split(3, dim = 1)
```
Comment (zhwesky2010) [inline tests/test_Tensor_tensor_split.py:61]: 加一个 result = a.tensor_split(3, 1) 的用法

File: paconvert/api_mapping.json [hunk 4]
```diff
@@ -1206,7 +1206,19 @@
       "memory_format"
     ]
   },
-  "torch.Tensor.dsplit": {},
+  "torch.Tensor.dsplit": {
+    "Matcher": "GenericMatcher",
+    "paddle_api": "paddle.Tensor.dsplit",
+    "min_input_args": 1,
+    "args_list": [
+      "indices",
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:1214]: 这个就写一个 吧，因为本质也只有一个参数
```
args_list": [
      "indices",
]
```

kwargs_change处理两个就行

File: paconvert/api_mapping.json [hunk 5]
```diff
@@ -1619,7 +1631,19 @@
     ]
   },
   "torch.Tensor.histogram": {},
-  "torch.Tensor.hsplit": {},
+  "torch.Tensor.hsplit": {
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:1634]: hsplit、dsplit问题一样

File: paconvert/api_mapping.json [hunk 6]
```diff
@@ -3405,7 +3460,21 @@
     "paddle_api": "paddle.Tensor.tanh_",
     "min_input_args": 0
   },
-  "torch.Tensor.tensor_split": {},
+  "torch.Tensor.tensor_split": {
+    "Matcher": "GenericMatcher",
+    "paddle_api": "paddle.Tensor.tensor_split",
+    "min_input_args": 1,
+    "args_list": [
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:3467]: 这个就写两个 吧，因为本质也只有两个参数，你这样写导致参数顺序挪位，会出问题
```
args_list": [
      "indices",
      "dim"
]
```

kwargs_change：
```
"kwargs_change": {
      "indices": "num_or_indices",
      "sections": "num_or_indices",
      "tensor_indices_or_sections": "num_or_indices",
      "dim": "axis"
    }
```
这样就能处理所有情况

**Comment by zhwesky2010** [comment]
infoflow 2024-07-16 11-12-16

映射文档的格式写得不对，这个要按实际代码来写，因为PyTorch的官网文档也可能描述不准，所以以实际代码为准。`torch.dsplit`这个API存在重载情况，需要将其分隔为两块来写，参考torch.max格式

**Comment by zhwesky2010** [comment]
@1147022743 请问这个什么时候修改

**Comment by 1147022743** [comment]
> @1147022743 请问这个什么时候修改

这周内把 我看完改一下

File: tests/test_Tensor_tensor_split.py (hunk 3 above)
Comment (1147022743) [reply tests/test_Tensor_tensor_split.py:61]: > 加一个 result = a.tensor_split(3, 1) 的用法

您好 这个用法运行结果与转换后的结果不同  猜测是Matcher的问题 但是该算子属于仅参数名不一致，使用的是通用的转换器，是否是映射关系判断错误了呢？
